### PR TITLE
New package: QSyncthingTray-0.5.8

### PR DESCRIPTION
**Pull request by thehowl**
Package for https://github.com/sieren/QSyncthingTray

### CODE REVIEW

**Comment by thehowl** [comment]
Hmm, I'm quite lost here. The builds fails on ARM - I presume that is because it can't fetch the necessary dependencies because they haven't been built for ARM?

File: srcpkgs/QSyncthingTray/template [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'QSyncthingTray'
+pkgname=QSyncthingTray
+version=0.5.8
+revision=1
+build_style=cmake
+makedepends="qt5-declarative-devel qt5-webengine-devel qt5-webchannel-devel qt5-location-devel qt5-quickcontrols2-devel qt5-devel"
```
Comment (Johnnynator) [inline srcpkgs/QSyncthingTray/template]: Pls create a new line after around ~70 chars.

File: srcpkgs/QSyncthingTray/template [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+short_desc="A OS X / Windows / Linux Tray App for Syncthing written in C++"
```
Comment (Johnnynator) [inline srcpkgs/QSyncthingTray/template]: Imo `Tray App for Syncthing written in C++` would be enough. Also never Start short_desc with a `A`

File: srcpkgs/QSyncthingTray/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+maintainer="Morgan Bazalgette <the@howl.moe>"
+license="LGPL-3.0"
```
Comment (Johnnynator) [inline srcpkgs/QSyncthingTray/template]: Pls use [SPDX Identifier ](https://spdx.org/licenses/)

File: srcpkgs/QSyncthingTray/template (hunk 2 above)
Comment (thehowl) [reply srcpkgs/QSyncthingTray/template]: Fair enough - I simply copied over the repo description, but I guess that works as well.

File: srcpkgs/QSyncthingTray/template (hunk 3 above)
Comment (thehowl) [reply srcpkgs/QSyncthingTray/template]: There is no clear indication whether the license is LGPL 3.0 only or LGPL 3.0 or later - should I assume it means "or later"?

Comment (Johnnynator) [reply srcpkgs/QSyncthingTray/template]: You have to take a look at the source files to determine it.
https://github.com/sieren/QSyncthingTray/blob/master/sources/qst/statswidget.cpp
The header there states `version 3.0 of the License, or (at your option) any later version.`

**Comment by Johnnynator** [comment]
Also pls do not save any icons or other binaries into git. Can't they generated or fetched at compile time?

**Comment by thehowl** [comment]
They are retrieved from the Syncthing repository: https://github.com/syncthing/syncthing/tree/master/assets

I'd avoid downloading the entire syncthing repo just to get the logos, and I'd prefer not to generate the pngs from the svg at compile time. Would a `wget` on `https://raw.githubusercontent.com/syncthing/syncthing/master/assets/logo-X.png` work?

**Comment by maxice8** [comment]
> Idk what the preferred way would be.

fetching with xbps-uhelper would be preferred for me

**Comment by Johnnynator** [comment]
```
  The imported target "Qt5::Core" references the file
     "/usr/lib/qt5/bin/moc"
```
This file is in `qt5-host-tools` so you additionally need it in hostmakedepends.

**Comment by thehowl** [comment]
aarch64 still fails. seems to be [a problem experienced by other PRs as well?](https://github.com/void-linux/void-packages/pulls?q=is%3Apr+xkeyboard-config+is%3Aclosed)

>ERROR: xkeyboard-config-2.24_1: the RSA signature is not valid!
ERROR: xkeyboard-config-2.24_1: removed pkg archive and its signature.

**Comment by Johnnynator** [comment]
The aarch64 issue can be ignored. Since it builds fine for `aarch64-musl` and `armv7l` it will most likely be also fine for `aarch64`. The problem is that someone messed something up and now there is a mismatch between the Signature and the Package (some others in the aarc64 repos are also affected)

**Comment by thehowl** [comment]
alright, should be good for review now then! Thanks for all the help @Johnnynator :)

File: srcpkgs/QSyncthingTray/template [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Template file for 'QSyncthingTray'
+pkgname=QSyncthingTray
+version=0.5.8
+revision=1
+build_style=cmake
+hostmakedepends="qt5-qmake qt5-host-tools"
+makedepends="qt5-declarative-devel qt5-webengine-devel qt5-webchannel-devel
+	qt5-location-devel qt5-quickcontrols2-devel qt5-devel"
```
Comment (Johnnynator) [inline srcpkgs/QSyncthingTray/template]: intend with only one space.

File: srcpkgs/QSyncthingTray/template [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+do_install() {
+	# Install main binary
+	vbin ./build/QSyncthingTray
+
+	# Desktop file
+	vmkdir usr/share/applications
+	vcopy ${FILESDIR}/qsyncthingtray.desktop usr/share/applications
+	# Download icons
+	cd ${FILESDIR}
```
Comment (Johnnynator) [inline srcpkgs/QSyncthingTray/template]: Nope. `$FILESDIR` does not get cleaned up automatically so you should not use it in the build process.
Just create a work folder for that inside `$wkrsrc`

**Comment by thehowl** [comment]
applied requested changes

**Pull request merged**